src/component/missions.js

```diff
@@ -1,7 +1,23 @@
-const Missions = () => (
-  <div>
-    missions
-  </div>
-);
+import { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import fetchMissions from '../fetchData';
+import { loadMissions } from '../redux/missions/missions';
+
+const Missions = () => {
+  const missions = useSelector((store) => store.missions);
+
+  const dispatch = useDispatch();
+
+  useEffect(async () => {
+    if (missions.length) return;
+    const missionsFetched = await fetchMissions();
+    dispatch(loadMissions(missionsFetched));
```

Comment:
Your payload should only contain `mission_id`, `mission_name`, and `description`

Reply:
Thanks for the suggestion @rtonata88 .

As you can see below, the reducer is actually doing that part, so the store only contains `mission_id`, `mission_name`, and `description`.

https://github.com/MacCrazyman/space-traveler-hub-react/blob/6bd7f99b59d34b761683bbde1a45fa1f6b735ac4/src/redux/missions/missions.js#L6-L10

Should I move the filter before the dispatch event?

Regards